Better AppStream metadata

AFCMS opened the pull request:
Fix #14647

## To do

This PR is Ready for Review

- [x] Rename `.appdata.xml` to `.metainfo.xml` ([.metainfo.xml is the recommended file name](https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#spec-component-location))
- [x] Import description translations from `.desktop`
- [x] Translate long description in french
- [x] Fix typo in french `.desktop` description
- [x] Use HTTPS URLs for screenshots
- [x] Add support info for mouse, keyboard, ~~gamepad~~ and touch
- [x] Add support info for supported screen size
- [x] Add support info for offline use 
- [x] Use the launchable tag to reference the `.desktop` file, remove `.desktop` from the top id (no idea why it was like that, didn't see it anywhere in the spec)
- [x] Add category and keywords
- [x] Add basic developer infos (is it OK?)
- [x] Fix URLs for `bugtracker` and `donation`
- [ ] Make first paragraph of long description longer? (>80 characters) Not a big deal IMO.

## How to test

Try running:

```sh
flatpak run --command=flatpak-builder-lint org.flatpak.Builder appstream ./misc/net.minetest.minetest.metainfo.xml
```

or this (less strict):

```sh
appstreamcli validate ./misc/net.minetest.minetest.metainfo.xml
```

> [!WARNING]
> With the renaming of the AppStream file, the Flathub manifest will need to be updated to reflect the change I think

sfan5 commented on misc/net.minetest.minetest.metainfo.xml:
```diff
@@ -0,0 +1,154 @@
+
+    <translation type="gettext">minetest</translation>
+
+    <update_contact>sfan5@live.de</update_contact>
```
Please use this opportunity to put someone else's email here.
Volunteers?

SmallJoker replied:
A Minetest mailing list would be convenient for such cases.

AFCMS commented:
I have added a value to display_length.

The goal is to make it appear as supported for both PC and mobile devices, so I picked 360 which is the value used by GNOME Calculator.

I have also added a tag to display the offline use of Minetest possible.

rubenwardy commented on misc/net.minetest.minetest.metainfo.xml:
```diff
@@ -0,0 +1,159 @@
+        <control>pointing</control>
+        <control>keyboard</control>
+        <control>touch</control>
+        <control>gamepad</control>
```
Minetest doesn't have functional support for gamepads, this should be removed until #4153 is done

AFCMS commented:
I consider this PR as finished, IMO the last TODO item isn't really important.

The last thing to check is the contact email, but don't know what to put here obviously.

sfan5 commented:
ruben and c55 are also listed here: https://github.com/minetest/minetest/security/policy
So one of them would be a good candidate.

AFCMS commented on misc/net.minetest.minetest.desktop:
```diff
@@ -1,10 +1,10 @@
 Comment[de]=Mehrspieler-Sandkastenspiel mit unendlichen Blockwelten
 Comment[es]=Juego sandbox multijugador con mundos infinitos
-Comment[fr]=Jeu multijoueurs de type bac à sable avec des mondes infinis
+Comment[fr]=Jeu multijoueur bac à sable avec des mondes infinis
```
Reverted

grorp commented on misc/net.minetest.minetest.metainfo.xml:
```diff
@@ -0,0 +1,158 @@
+    <summary xml:lang="fr">Jeu multijoueur bac à sable avec des mondes infinis</summary>
+    <summary xml:lang="ja">マルチプレイに対応した、無限の世界のブロック型サンドボックスゲームです</summary>
+    <summary xml:lang="ru">Игра-песочница с безграничным миром, состоящим из блоков</summary>
+    <summary xml:lang="tr">Tek-Çok oyuncuyla küplerden sonsuz dünyalar inşa et</summary>
```
These descriptions describe Minetest as a game, which is not what Minetest is today. It's an engine, or even more correctly a platform. If you're already at it, it would make sense to write a new description.

Recently I also replaced the description on the Google Play Store.

AFCMS replied:
Done!

Zughy commented:
@AFCMS updates?

okias commented:
ACK on appdata/metainfo transition. Look reasonable with requested changes.

AFCMS commented:
Updated the descriptions to mention the fact that Minetest is a game platform.

English and French should be good, for other translations I just used DeepL and can't really validate those.

If some people with knowledge of the respective languages could verify the translations that would be great.

okias commented:
I guess @celeron55 would be good choice, as @sfan5 pointed https://github.com/minetest/minetest/security/policy

okias commented on misc/net.minetest.minetest.metainfo.xml:
```diff
@@ -0,0 +1,158 @@
+    <summary xml:lang="tr">Çok oyunculu sonsuz dünya blok sandbox oyun platformu</summary>
+
+    <metadata_license>CC0-1.0</metadata_license>
+    <project_license>LGPL-2.1+ and CC-BY-SA-3.0 and MIT and Apache-2.0</project_license>
```
uppercase `AND`, please

AFCMS commented:
Updated the contact email to c55

grorp commented:
I still don't like the short description tbh. It's a lot of words which don't really make sense together.

> Multiplayer infinite-world block sandbox game platform

Minetest is ... what? An infinite-world ... or an infinite-world block sandbox ... or a sandbox game ... or a game platform? Reduced version with more focus:

> Block-based multiplayer game platform

Also, regarding the German translation:

> Mehrspieler-Sandkasten-Spielplattform für unendliche Welten

This highlights the "infinite worlds" aspect too much.

> Minetest ist eine unendliche Blocksandkasten-Spielplattform.

"Blocksandkasten-Spieleplattform"? What's that?

grorp commented on misc/net.minetest.minetest.metainfo.xml:
```diff
@@ -0,0 +1,158 @@
+
+    <description>
+        <p>
+            Minetest is a block based sandbox game platform.
```
```suggestion
            Minetest is a block-based sandbox game platform.
```

sfan5 commented on misc/net.minetest.minetest.desktop:
```diff
@@ -1,13 +1,13 @@
+Comment[de]=Blockbasierte Multiplayer-Spielplattform
+Comment[es]=Plataforma de juegos multijugador basada en bloques
+Comment[fr]=Plate-forme de jeu multijoueurs à base de blocs
+Comment[ja]=ブロックベースのマルチプレイヤーゲームプラットフォーム
```
are these google translate?

AFCMS replied:
Outside of french, yes (actually DeepL).

If some people with knowledge of these languages could check that would be great.

sfan5 replied:
The German one unnecessarily uses an anglicism and while the Japanese one is correct (afaict) it's a loanword soup.

I'd rather remove these and wait for native speakers to add correct translations.
